src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxDelegate.iOS.cs

```diff
@@ -39,8 +39,16 @@ public override bool ShouldChangeCharacters(UITextField textField, NSRange range
 				{
 					var newLength = textBoxView.Text.Length + replacementString.Length - range.Length;
 					return newLength <= _textBox.GetTarget()?.MaxLength;
-				};
-			}
+				};
+
+				if (_textBox.GetTarget() is not TextBox textBox)
+				{
+					return false;
+				}
+
+				// Both IsReadOnly = true and IsTabStop = false can prevent editing
+				return !textBox.IsReadOnly && textBox.IsTabStop;
+			}
 
 			return true;
```

Comment:
I'm wondering if we should fallback to `base.ShouldChangeCharacters(...)` or it's find to just return `true`. Do you know in what cases `base.ShouldChangeCharacters` will return false?

Reply:
`base.ShouldChangeCharacters` is an optional delegate method and these usually do not add any functionality inside it. Xamarin uses this pattern to remove the hassle from the developers to implementing the `IUITextFieldDelegate` interface.

The default value Xamarin returns is `true`.